airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -88,12 +88,16 @@ public AirbyteConnectionStatus check(final JsonNode config) {
       final BigQuery bigquery = getBigQuery(config);
       final UploadingMethod uploadingMethod = BigQueryUtils.getLoadingMethod(config);
 
+      // check has permission for custom dataset creation (may be required for normalization)
       BigQueryUtils.checkHasCreateAndDeleteDatasetRole(bigquery, datasetId, datasetLocation);
 
       final Dataset dataset = BigQueryUtils.getOrCreateDataset(bigquery, datasetId, datasetLocation);
       if (!dataset.getLocation().equals(datasetLocation)) {
         throw new ConfigErrorException("Actual dataset location doesn't match to location from config");
       }
+      // check permission for provided dataset
+      BigQueryUtils.attemptCreateTableAndTestInsert(bigquery, datasetId);
```

Comment:
Isn't this method already called in `checkHasCreateAndDeleteDatasetRole`? How come we're re-using this method after it's already called above?

Reply:
+1. `checkHasCreateAndDeleteDatasetRole` calls `attemptCreateTableAndTestInsert`, so I am not sure what adding it here does

Reply:
Yup. It's, but the first time we try to create a new table and insert it to a newly created schema meanwhile in this PR I added calls to check it in the selected schema itself

Reply:
I don't think I understand how that is so. This line `BigQueryUtils.attemptCreateTableAndTestInsert(bigquery, datasetId); ` uses the same `datasetId` as this line `BigQueryUtils.checkHasCreateAndDeleteDatasetRole(bigquery, datasetId, datasetLocation);`, so how are they going to be checking different schemas? AFAIK, `config` object that is passed to `check` method contains the default destination configuration and if the user chooses "Mirror source structure" or "Custom format", this datasetId is not going to be the same datasetId as the one where the destination will attempt writing during the sync job. What am I missing here?

Reply:
In the already existing verification, we add a suffix 
 final String tmpTestDatasetId = datasetId + CHECK_TEST_DATASET_SUFFIX  + System.currentTimeMillis();

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryUtils.java#L128

So basically in the master branch we currently don't try to write anything directly to the schema set by user in connection config. But we create another schema and do all actions in this tmp schema. 

In this PR I added logic to also create tmp table and write to it directly to the schema set by user. So more cases would be tested now

Reply:
> In this PR I added logic to also create tmp table and write to it directly to the **schema set by user**. So more cases would be tested now.

I don't see how this part is true:  **schema set by user**. What this PR does if I read it correctly is try to create a dataset with a random name and then a test table in that dataset. I don't see where it uses a dataset specified by user in replication configuration. Specifically, if a user choses "Mirror source structure" option, this test will not be trying to create tables in the same namespace where the sync will try to create tables.